bot/exts/core/leaderboard.py

```diff
@@ -0,0 +1,206 @@
+        """Check the interactor is authorised."""
+        if interaction.user.id == self.authorization:
+            return True
+
+        await interaction.response.send_message(
+            ":no_entry_sign: You are not authorized to perform this action.",
+            ephemeral=True,
+        )
+
+        return False
+
+    @ui.button(label="Confirm", style=ButtonStyle.green, row=0)
+    async def confirm(self, _button: ui.Button, interaction: Interaction) -> None:
+        """Redeploy the specified service."""
+        for global_lb, _ in self.bot.games_leaderboard.values():
+            await global_lb.clear()
+
+        log.info(f"The leaderboard was cleared by Member({self.authorization})")
+        await interaction.response.send_message(
+            content=":white_check_mark: Cleared all game leaderboards.",
+            ephemeral=False
+        )
+
+        self.stop()
+
+    @ui.button(label="Cancel", style=ButtonStyle.grey, row=0)
+    async def cancel(self, _button: ui.Button, interaction: Interaction) -> None:
+        """Logic for if the deployment is not approved."""
+        await interaction.response.send_message(
+            content=":x: Clearing cache aborted!",
+            ephemeral=False,
+        )
+        self.stop()
+
+
+class Leaderboard(commands.Cog):
+    """Cog for getting game leaderboards."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @staticmethod
+    def get_rank(n: int) -> str:
+        """Get the rank in form of emoji (first 3 places) or ordinal number for `n`."""
+        first_3 = {1: ":first_place:", 2: ":second_place:", 3: ":third_place:"}
+        if n in first_3:
+            return first_3[n]
+
+        suffix = ["th", "st", "nd", "rd", "th"][min(n % 10, 4)]
+        if 11 <= (n % 100) <= 13:
+            suffix = "th"
+        return str(n) + suffix
+
+    async def make_leaderboard(
+        self,
+        author: discord.Member,
+        cached_leaderboard: list[RedisCache]
+    ) -> discord.Embed:
+        """Make a discord embed for the current top 10 members in the cached leaderboard."""
+        if len(cached_leaderboard) == 1:
+            game_leaderboard = await cached_leaderboard[0].to_dict()
+            leaderboard = Counter(game_leaderboard)
+        else:
+            leaderboard = Counter()
+            for lb in cached_leaderboard:
+                game_leaderboard = await lb.to_dict()
+                leaderboard += Counter(game_leaderboard)
+
+        top_ten = leaderboard.most_common(10)
+        try:
+            author_rank = list(leaderboard.keys()).index(author.id) + 1
+        except ValueError:
+            author_rank = 0
+
+        lines = []
+        for index, (member_id, score) in enumerate(top_ten, start=1):
+            mention = f"<@{member_id}>"
+            rank = self.ordinal_number(index)
+            buffer = 4 if index >= 4 else len(rank)+1
+            formatted_rank = format(rank, f"\u2800<{buffer}")
```

Comment:
What's that unicode?

Reply:
It is a [different type of space](https://www.compart.com/en/unicode/U+2800), using the normal space (` `) didn't keep the items aligned in the embed and they were stripped out.

Reply:
Can you switch this to the name of the character? See [this page](https://docs.python.org/3/howto/unicode.html#the-string-type) and scroll down to:
```python
>>> "\N{GREEK CAPITAL LETTER DELTA}"  # Using the character name
'\u0394'
>>> "\u0394"                          # Using a 16-bit hex value
'\u0394'
>>> "\U00000394"                      # Using a 32-bit hex value
'\u0394'
```